src/qibo/ui/mpldrawer.py

```diff
@@ -0,0 +1,721 @@
+        fill=False,
+        lw=plot_style["linewidth"],
+        label="",
+    )
+    ax.add_patch(rect)
+
+
+def _get_flipped_indices(targets, labels):
+    return [_get_flipped_index(t, labels) for t in targets]
+
+
+def _render_label(label, inits={}):
+    """Slightly more flexible way to render labels.
+
+    >>> _render_label('q0')
+    '$|q0\\\\rangle$'
+    >>> _render_label('q0', {'q0':'0'})
+    '$|0\\\\rangle$'
+    """
+    if label in inits:
+        s = inits[label]
+        if s is None:
+            return ""
+        else:
+            return r"$|%s\rangle$" % inits[label]
+    return r"$|%s\rangle$" % label
+
+
+def _make_cluster_gates(gates_items):
+
+    temp_gates = []
+    temp_mgates = []
+    cluster_gates = []
+
+    for item in gates_items:
+        if len(item) == 2:  # single qubit gates
+            if item[0] == "MEASURE":
+                temp_mgates.append(item)
+            else:
+                if len(temp_gates) > 0:
+                    if item[1] in [tup[1] for tup in temp_gates]:
+                        cluster_gates.append(temp_gates)
+                        temp_gates = []
+                        temp_gates.append(item)
+                    else:
+                        temp_gates.append(item)
+                else:
+                    temp_gates.append(item)
+        else:
+            if len(temp_gates) > 0:
+                cluster_gates.append(temp_gates)
+                temp_gates = []
+
+            if len(temp_mgates) > 0:
+                cluster_gates.append(temp_mgates)
+                temp_mgates = []
+
+            cluster_gates.append([item])
+
+    if len(temp_gates) > 0:
+        cluster_gates.append(temp_gates)
+
+    if len(temp_mgates) > 0:
+        cluster_gates.append(temp_mgates)
+
+    temp_gates = []
+    temp_mgates = []
+
+    return cluster_gates
+
+
+def _build_path(filename):
+    file_path = path.abspath(__file__)  # full path of current file
+    dir_path = path.dirname(file_path)  # full path of the directory from file
+    final_path = path.join(dir_path, filename)  # absolute file path of given file
+    return final_path
+
+
+def _check_list_str(list, str):
+    for item in list:
+        if item in str:
+            return True
+    return False
+
+
+def _process_gates(array_gates):
+    gates_plot = []
+
+    for gate in array_gates:
+        init_label = gate.name.upper()
+
+        if init_label == "CCX":
+            init_label = "TOFFOLI"
+
+        if init_label == "CX":
+            init_label = "CNOT"
+
+        if _check_list_str(["SX", "CSX"], init_label):
+            is_dagger = init_label[-2:] == "DG"
+            init_label = (
+                r"$\rm{\sqrt{X}}^{\dagger}$" if is_dagger else r"$\rm{\sqrt{X}}$"
+            )
+
+        if (
+            len(gate._control_qubits) > 0
+            and "C" in init_label[0]
+            and "CNOT" != init_label
+        ):
+            init_label = gate.draw_label.upper()
```

Comment:
I guess they are all `elif`, since if one is true, the other ones are not. No need to repeat the checks if you already matched one of the conditions